### ISSUE TITLE
[Verification] Be able to define provider state without adding a stateChangeEndpoint

### ISSUE THREAD

**Issue by gaeljw** [jbwheatley/pact4s#121]
Would there be a way to not having to add a custom endpoint only for the tests in order to setup the provider state when verifying pacts on the provider side?

With Pact-JVM and Junit, it's only a matter of a dedicated method that runs to setup the provider state.

I believe it would be way easier to do it in a similar way in pact4s. I'm not sure adding an endpoint only for tests is an easy task in all web frameworks and that could discourage people to use pact4s.

**Comment by jbwheatley** [comment]
This may be easier said than done, because the running of individual interactions is managed by `pact-jvm`, so we can only easier set-up state through the api that pact-jvm provides - which is by adding a state change endpoint

**Comment by jbwheatley** [comment]
we may be able to hack around it converting a user defined `f: String => Unit` that we turn into a http server that `pact-jvm` sends requests to, instead of a user-defined endpoint

**Comment by gaeljw** [comment]
Okay maybe we need to dig into how pact-jvm does its integration with junit.

From https://github.com/pact-foundation/pact-jvm/tree/master/provider/junit it looks as a simple method call but under the hood maybe it does something like what you suggest.

```java
        @State({"default", "no-data"}) // Method will be run before testing interactions that require "default" or "no-data" state
        public void toDefaultState() {
            // Prepare service before interaction that require "default" state
            // ...
            System.out.println("Now service in default state");
        }
```

**Comment by jbwheatley** [comment]
My guess (without looking) is there is some reflection going on..

**Comment by jbwheatley** [comment]
released in v0.1.0
